store/types/gas.go

```diff
@@ -69,9 +71,19 @@ func (g *basicGasMeter) GasConsumedToLimit() Gas {
 	return g.consumed
 }
 
+// Copied from types/int.go
+// AddUint64Overflow performs the addition operation on two uint64 integers and
+// returns a boolean on whether or not the result overflows.
+func AddUint64Overflow(a, b uint64) (uint64, bool) {
```

Comment:
Why are we copying this over? Import-cycles?

Reply:
Yes. Seems like `AddUint64Overflow` is only being used by the gas code(at least currently), it might be better to delete on `types/` side.

Reply:
All store related files (`gas.go`, `store.go`, etc) will be moved to `store/types/` and the core interface (`KVStore, etc.`) will be reexported on `types/`, because of the cyclic dependency problem. The remaining part in `store/` that using `types/` will be cleared on the second part.